subprojects/gradle/design-screenshots/src/main/kotlin/com/avito/plugin/DeviceProvider.kt

```diff
@@ -0,0 +1,50 @@
+package com.avito.plugin
+
+import com.avito.runner.service.worker.device.DeviceCoordinate
+import com.avito.runner.service.worker.device.DevicesManager
+import com.avito.runner.service.worker.device.Serial
+import com.avito.runner.service.worker.device.adb.Adb
+import com.avito.runner.service.worker.device.adb.AdbDevice
+import com.avito.runner.service.worker.device.adb.AdbDeviceParams
+import com.avito.utils.logging.CILogger
+import com.avito.utils.logging.commonLogger
+
+internal interface DeviceProvider {
+
+    fun getDevice(): AdbDevice
+}
+
+internal class DeviceProviderLocal(
+    private val adb: Adb,
+    private val adbDevicesManager: DevicesManager,
+    private val logger: CILogger
+) : DeviceProvider {
+
+    override fun getDevice(): AdbDevice {
+        val adbDeviceParams = connectedDeviceParams()
+
+        val serial: Serial.Local = when {
+            adbDeviceParams.id is Serial.Local -> adbDeviceParams.id as Serial.Local
+            adbDeviceParams.id is Serial.Remote ->  Serial.Local(adbDeviceParams.id.value) // TODO: support model of locally connected device
```

Comment:
Didn't change in this PR, but it seems a valid case.
It's helpful to connect emulator from local Docker. Especially for screenshots test.

Reply:
As I know, screenshot generating works only at the `Local emulator`

Reply:
The term "local" is confusing. 
I mean local emulator inside Docker container. It is connected to adb as "localhost:5555".
This is the easiest way for me to use screenshots tests locally.

Reply:
As I know @DEADMC made screenshots by the emulator without docker on the local machine